### PR TITLE
Fix ValueError due to unmatched

### PR DESCRIPTION
**Pull request by sunset1995**
During my offline evaluation, I encounter
```
ValueError: cost matrix is infeasible
``` 
which is caused by unmatched cost matrix.

An illustration.
螢幕快照 2020-07-02 上午12 13 31

I suggest using `maximum_bipartite_matching` instead.

### CODE REVIEW

**Comment by bertjiazheng** [comment]
@sunset1995 Thanks for your PR. Would you mind providing a test case for this?

**Comment by sunset1995** [comment]
Sure my pleasure.

This example is from the original test split -- `scene_03308_1034.txt`.
I found the provided annotation not following the new format (occlusion part still existed in the ground-truth), so I have re-project them myself.

```
(Pdb) pp annotations
array([[ 53, 163],
       [ 53, 354],
       [470, 180],
       [470, 337],
       [470, 200],
       [470, 316],
       [489, 199],
       [489, 317],
       [495, 211],
       [495, 303],
       [543, 212],
       [543, 303],
       [552, 200],
       [552, 315],
       [645, 216],
       [645, 299],
       [845, 207],
       [845, 308]])
(Pdb) pp predictions
array([[ 52, 164],
       [ 52, 356],
       [471, 181],
       [471, 338],
       [475, 180],
       [475, 338],
       [491, 211],
       [491, 305],
       [550, 212],
       [550, 304],
       [563, 199],
       [563, 318],
       [648, 215],
       [648, 300],
       [844, 208],
       [844, 309]])
```

The corresponding visualization.
scene_03308_1034

I think such error occur whenever there are two GT corner but only one predicted corner matched.

**Comment by bertjiazheng** [comment]
First of all, Thanks for your PR!

I found https://github.com/scipy/scipy/issues/6900 is related to this problem. Following https://github.com/scipy/scipy/issues/6900#issuecomment-451735634, I replace the `np.inf` with a finite distance value (such as the diagonal distance of the image) and found it worked. 

In your PR, you propose to use `maximum_bipartite_matching` to replace the `linear_sum_assignemnt`. It is slightly different between these two solvers as `linear_sum_assignemnt` takes the distance values into account. This may not affect the final measurements but result in the different matched pairs.

I tested the given case and got the same F-score using these two solvers. The differences are the two matched pairs: the former select the pair (3, 3) and (11, 11), and the latter select the pair (2, 3) and (10, 11). However, the `dist(3, 3) = 4, dist(11, 11) = 4.47213595499958`. `dist(3, 3) = 12.165525060596439, dist(11, 11) = 12.165525060596439`.

In summary, I prefer to replace `np.inf` with finite distance value, and still use `linear_sum_assignemnt`. If you feel comfortable about this, let me fix this bug.

**Comment by sunset1995** [comment]
Sure ;).  Thanks for the fix.
